File: test_client.py
from flask import Flask, render_template, request, redirect, url_for, jsonify
import requests, json
import os
import shutil


app = Flask(__name__)

def send_post():
    file = {'date': 2, 'name': 'my first post', 'text': 'Hey! This is my first post to test the system.'}
    with open('data.json', 'w') as output:
        json.dump(file, output)
    files = {'file': 'data.json'}
    values = {'id': '1'}
    r = requests.post('http://localhost:5000/add_post', files=files, headers=values)


def view_posts():
    values = {'id': '1'}
    r = requests.post('http://localhost:5000/view_all_posts', headers=values)


send_post()
view_posts()

File: test_client.js
const id = 1
const url = 'http://localhost:5000/users'
const url2 = 'http://localhost:5000/add_post'
const request = new XMLHttpRequest()
request.open('GET', url, false)
request.setRequestHeader('id', "1")
    // Send request
request.send()
JSON.parse(request.responseText).forEach(user => {
    if(id == user['id']){
        const newData = {
			date: 2,
            title: "My first post",
            text: 'Hey! This is my first post to test the system.'
			}
		// user['posts'].push(newData)

        //send new data to API
        fetch(url2, {
            method: 'PATCH',
            body: JSON.stringify(newData),
            headers:
                {
                    "Content-type": "application/json; charset=UTF-8",
                    'Authorization': `Token ${token}`
                }
        })

    }
})
